src/main/java/rose/command/AddCommand.java

```diff
@@ -0,0 +1,82 @@
+package rose.command;
+
+import rose.RoseException;
+import rose.Storage;
+import rose.TaskList;
+import rose.Ui;
+import rose.task.*;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a command used by user to add task to the list.
+ * <p>An <code>AddCommand</code> object is represented by its task type and description of the task.
+ */
+public class AddCommand extends Command {
+    private final TaskType taskType;
+    private final String taskName;
+
+    public AddCommand(TaskType taskType, String taskName) {
+        this.taskType = taskType;
+        this.taskName = taskName;
+    }
+
+    /**
+     * Adds a new task to the list.
+     *
+     * @param tasks current list of tasks.
+     * @param ui ui object to show message to user.
+     * @param storage storage object to store the data.
+     * @throws RoseException If input is incomplete or tasktype is unknown.
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        Task newTask;
+        DateTimeFormatter INPUT_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+        try {
+            switch (taskType) {
+            case TODO:
+                newTask = new Todo(taskName);
+                break;
+            case DEADLINE:
+                String[] deadlineParts = taskName.split(" /by ");
+                if (deadlineParts.length < 2) {
+                    throw new RoseException("Deadline task is missing '/by'.");
+                }
+                newTask = new Deadline(deadlineParts[0], LocalDate.parse(deadlineParts[1], INPUT_FORMAT));
+                break;
+            case EVENT:
+                String[] eventPartsA = taskName.split(" /from ");
```

Comment:
Maybe you can consider renaming variables to be clearer.
`eventPartsA` to `fromParts`
`eventPartsB` to `toParts`